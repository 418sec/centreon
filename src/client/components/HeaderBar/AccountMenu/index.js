export { default as AccountMenu } from './AccountMenuContainer';