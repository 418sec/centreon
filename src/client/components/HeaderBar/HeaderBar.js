import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { withStyles } from 'material-ui/styles'
import Typography from 'material-ui/Typography'
import AppBar from 'material-ui/AppBar'
import Toolbar from 'material-ui/Toolbar'
import Tabs, { Tab } from 'material-ui/Tabs'
import { Link } from "react-router-dom"

const styles = theme => ({
  appBar: {
    position: 'inherit',
    backgroundColor: '#009fdf'
  },
  toolBar: {
    display: 'flex',
    justifyContent: 'space-between'
  },
  title: {
    marginRight: "40px"
  },
  tabs: {
    flex: 1
  },
  tab: {
    height: '64px'
  }
})

class HeaderBar extends Component {

  render() {
    const { classes, name, entries, value, handleChange, children } = this.props

    return (
      <AppBar className={classes.appBar}>
        <Toolbar className={classes.toolBar}>
          <Typography type="title" color="inherit" className={classes.title}>
            {name}
          </Typography>
          {entries && entries.length &&
            <Tabs
              value={value}
              onChange={handleChange}
              className={classes.tabs}
              indicatorColor="primary"
            >
              {entries.map((entry, index) => {
                return (
                  <Tab
                    className={classes.tab}
                    key={"tab" + index}
                    label={entry.name}
                    component={Link}
                    to={entry.path}
                  />
                )
              })}
            </Tabs>
          }
          <div>
            {children}
          </div>
        </Toolbar>
      </AppBar>
    )
  }
}

HeaderBar.propTypes = {
  classes: PropTypes.object.isRequired,
  theme: PropTypes.object.isRequired,
};

export default withStyles(styles, { withTheme: true })(HeaderBar)