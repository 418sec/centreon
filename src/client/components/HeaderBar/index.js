export { default as HeaderBar } from './HeaderBarContainer';