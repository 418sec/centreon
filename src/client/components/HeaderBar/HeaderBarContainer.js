import React, { Component } from 'react'
import { connect } from 'react-redux'
import HeaderBar from './HeaderBar'
import { AccountMenu } from 'Components/HeaderBar/AccountMenu'
import { withRouter } from 'react-router'

export class HeaderBarContainer extends Component {

  constructor (props) {
    super(props)

    const { location, entries } = props
    const path = location.pathname

    let value = 0
    if (entries && entries.length) {
      entries.map((entry, index) => {
        if (path == entry.path) {
          value = index
        }
      })
    }

    this.state = {
      value: value
    }
  }

  handleChange = (event, value) => {
    this.setState({ value })
  }

  render() {
    const { title, entries } = this.props
    const { value } = this.state

    return (
      <HeaderBar
        name={title}
        entries={entries}
        value={value}
        handleChange={this.handleChange}
      >
        <AccountMenu/>
      </HeaderBar>
    )
  }
}

const mapStateToProps = (store) => {
  return {
    title: store.headerReducer.title,
    entries: store.headerReducer.entries
  }
}

export default withRouter(connect(mapStateToProps)(HeaderBarContainer))