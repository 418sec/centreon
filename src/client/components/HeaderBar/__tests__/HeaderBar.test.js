/* eslint-env jest */
import React from 'react'
import Enzyme from 'enzyme'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import { HeaderBarContainer as HeaderBar } from 'Components/HeaderBar/HeaderBarContainer'

Enzyme.configure({ adapter: new Adapter() })

describe('HeaderBar', () => {
  let component
  const props = {
    location: {
      pathname: '/app/instances'
    },
    title: '',
    entries: {}
  }

  it('renders correctly', () => {
    component = shallow(
      <HeaderBar {...props}/>
    ).dive().dive()
    expect(toJson(component)).toMatchSnapshot()
  })
})