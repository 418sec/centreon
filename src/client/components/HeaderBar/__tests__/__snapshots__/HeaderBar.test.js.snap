// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBar renders correctly 1`] = `
<WithStyles(AppBar)
  className="HeaderBar-appBar-1"
>
  <WithStyles(Toolbar)
    className="HeaderBar-toolBar-2"
  >
    <WithStyles(Typography)
      className="HeaderBar-title-3"
      color="inherit"
      type="title"
    />
    <div>
      <Connect(AccountMenuContainer) />
    </div>
  </WithStyles(Toolbar)>
</WithStyles(AppBar)>
`;
